### PR TITLE
Clarify CDR training docs regarding the use of a markov chain monte carlo

### PR DESCRIPTION
**Pull request by natestemen**
## Description

There is an important omission from our documentation regarding how Mitiq creates training circuits for Clifford Data Regression. There are two methods put forth in the [original paper](https://arxiv.org/abs/2005.10189), yet the authors choice is different from the one in Mitiq. This PR adds clarification for that detail.

- bddae0b0a0fdab8df2483a17b3544d90831668ec contains general cleanup of the CDR pages
- 5abe596a4cacafbf4b0d8e17d45995c6a889f8f9 is the issue that "resolves" this issue.

fixes https://github.com/unitaryfund/mitiq/issues/1393

### CODE REVIEW

File: docs/source/guide/cdr-5-theory.md [hunk 1]
```diff
@@ -13,30 +13,16 @@ kernelspec:
...
+3. **Learn the Error Mitigation Model.** A model $f(X^{\text{error}}, a)$ for $X^{exact}$ is defined and learned. Here, $a$ is the set of parameters to be determined. This is achieved by minimizing the distance between the training set, as expressed by the following optimization problem: $a_{opt} = \underset{a}{\text{argmin}} \sum_i \left| X_{\phi_i}^{\text{exact}} - f(X_{\phi_i}^{\text{error}},a) \right|^2.$ In this expression, $a_{opt}$ are the parameters that minimize the cost function.
+4. **Apply the Error Mitigation Model.** Finally, the learned model $f(X^{\text{error}}, a_{opt})$ is used to correct the expectation values of $X$ for new quantum states, expressed as $X_\psi^{\text{exact}} = f(X_\psi^{\text{error}}, a_{opt})$.
 
+The effectiveness of this method has been proven on circuits with up to 64 qubits and for tasks such as estimating ground-state energies.
```
Comment (Misty-W) [inline docs/source/guide/cdr-5-theory.md:27]:
```suggestion
The effectiveness of this method has been demonstrated on circuits with up to 64 qubits and for tasks such as estimating ground-state energies.
```

File: docs/source/guide/cdr-5-theory.md [hunk 2]
```diff
@@ -13,30 +13,16 @@ kernelspec:
...
 
-Clifford data regression (CDR) is a quantum error mitigation technique that has been introduced in Ref. {cite}`Czarnik_2021_Quantum` and extended to variable-noise CDR in Ref. {cite}`Lowe_2021_PRR`.
-. The presented error mitigation (EM) strategy is designed for gate-based quantum computers. This method primarily consists of creating a training data set $\{(X_{\phi_i}^{\text{error}}, X_{\phi_i}^{\text{exact}})\}$, where $X_{\phi_i}^{\text{error}}$ and $X_{\phi_i}^{\text{exact}}$ are the expectation values of an observable $X$ for a state $|\phi_i\rangle$ under error and error-free conditions, respectively. 
+Clifford Data Regression is a quantum error mitigation technique introduced in {cite}`Czarnik_2021_Quantum` and extended to variable-noise CDR in {cite}`Lowe_2021_PRR`.
+The error mitigation strategy is designed for gate-based quantum computers.
+This method primarily consists of creating a training data set $\{(X_{\phi_i}^{\text{error}}, X_{\phi_i}^{\text{exact}})\}$, where $X_{\phi_i}^{\text{error}}$ and $X_{\phi_i}^{\text{exact}}$ are the expectation values of an observable $X$ for a state $|\phi_i\rangle$ under error and error-free conditions, respectively.
```
Comment (Misty-W) [inline docs/source/guide/cdr-5-theory.md:18]: I think the following flows slightly more naturally.
```suggestion
This error mitigation strategy is designed for application at the gate level and is relatively straightforward to implement on gate-based quantum computers.
CDR primarily consists of creating a training data set $\{(X_{\phi_i}^{\text{error}}, X_{\phi_i}^{\text{exact}})\}$, where $X_{\phi_i}^{\text{error}}$ and $X_{\phi_i}^{\text{exact}}$ are the expectation values of an observable $X$ for a state $|\phi_i\rangle$ under error and error-free conditions, respectively.
```

**Review by Misty-W — APPROVED** [review]
Nice work @natestemen!

A few minor suggestions, otherwise LGTM!

File: docs/source/guide/cdr-4-low-level.md [hunk 3]
```diff
@@ -25,9 +23,19 @@ name: cdr-workflow
...
+In CDR, the generation of quantum circuits is different, as it involves the generation of training circuits.
+
+```{warning}
+In {cite}`Czarnik_2021_Quantum`, the authors lay out two different methods for generating the training circuits.
+
+1. Randomly replacing gates in the target circuit with nearby Clifford gates.
+2. Construct new circuits with the use of a Markov Chain Monte Carlo (MCMC) which produce classicaly simulable states.
 
-The CDR workflow Figure above shows a schema of the implementation of CDR in Mitiq. Similarly to ZNE and PEC, also CDR in Mitiq is divided in two main stages: The first one of circuit generation and the second for inference of the mitigated value. However, in CDR, the generation of quantum circuits is different, as it involves the generation of training circuits. The division of CDR into training, learning and prediction stages is shown in the figure below.
+The authors of {cite}`Czarnik_2021_Quantum` derive results with the use of the MCMC method, whereas Mitiq uses simpler approach presented in point 1.
```
[the remark this answers is not in the document]
Comment (natestemen) [reply docs/source/guide/cdr-4-low-level.md:35]: To my knowledge there has been no studies of how the circuit generation procedure effects the performance of CDR. Would be nice though!

**Pull request merged**